### PR TITLE
Build ocs-crossbar image from official crossbario/crossbar image

### PR DESCRIPTION
**Pull request by BrianJKoopman**
## Description

Instead of building our own crossbar image by installing crossbar on an ocs base image, use the [official crossbario/crossbar](https://hub.docker.com/r/crossbario/crossbar) image from Dockerhub. This bundles a default config generated by ocsbow into the image, which should mostly eliminate the need for test institutions to generate and mount a crossbar config.

Also update the documentation quite a bit to describe this change and how to mount a new config into the container/rebuild if needed. I got a little carried away to documentation work, as this also includes updates to instructions that describe how to split out various components to run separate from the ocs docker-compose.yml file so we can avoid always bringing down services like influxdb/crossbar/grafana.

## Motivation and Context

The build of crossbar we were making was broken. Rather than fix it I thought it best to switch to the official image, or at least one based on the official image. This uses a `crossbar` user internally, so has some small hurdles if mounting a config file, but the way to jump over those is described in the new docs.

## How Has This Been Tested?

I have built and run the new crossbar image locally in a small test environment.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] Unless I am preparing a release, I have opened this PR onto the `develop` branch.

### CODE REVIEW

File: docs/user/crossbar_config.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+For further details on crossbar server configuration, see the crossbar `Router
+Configuration`_ page.
+
+.. _`ocs/docker/crossbar/config.json`: https://github.com/simonsobs/ocs/blob/develop/docker/crossbar/config.json
+.. _`ocs/ocs/support/crossbar_config.json`: https://github.com/simonsobs/ocs/blob/develop/ocs/support/crossbar_config.json
+.. _`Router Configuration`: https://crossbar.io/docs/Router-Configuration/
+
+Generating a New Config File
+----------------------------
+``ocsbow`` can be used to generate a default configuation file, based on
```
Comment (mhasself) [inline docs/user/crossbar_config.rst:39]: typo

File: docs/user/crossbar_config.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+You should then update your configuration to use the new, local,
+``ocs-crossbar`` image.
+
+Bind Mounting the Configuration
+-------------------------------
+To instead mount the new configuration into the pre-built image, first chown
+your file to be owned by user and group 242 (the default crossbar UID/GID),
+then mount it appropriately in your docker-compsose file. Here we assume you
+put the configuration in the direcitory ``./dot_crossbar/``::
```
Comment (mhasself) [inline docs/user/crossbar_config.rst:107]: typo

File: docs/user/crossbar_config.rst [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    $ chown -R 242:242 dot_crossbar/
```
Comment (mhasself) [inline docs/user/crossbar_config.rst:109]: Can we explore other ways of doing this?  It looks like there are ways to map a host system uid/gid to a container uid/gid.  User/group 242 on the host system could be anything, and it's odd to have files lying around with bizarre ids like this.

https://linuxnatives.net/2019/align-user-ids-inside-and-outside-docker-with-subuser-mapping

**Review by mhasself — APPROVED** [review]
Thanks for this.

Fixing the chown thing shouldn't hold this up but I think we should figure out if there's a better way to handle permissions on mounted volumes going forward.

File: docs/user/crossbar_config.rst (hunk 3 above)
Comment (BrianJKoopman) [reply docs/user/crossbar_config.rst:109]: I'll definitely look into this. I wasn't crazy about the chown solution. That was part of the reason I avoided using this base image in the first place. It's worth figuring out though, so we can run as a non-root user within the containers.

**Comment by BrianJKoopman** [comment]
> Thanks for this.
> 
> Fixing the chown thing shouldn't hold this up but I think we should figure out if there's a better way to handle permissions on mounted volumes going forward.

Great, I'll merge now then and open an issue about the permissions problem.

**Pull request merged**